plutus-playground-client/src/MainFrame/State.purs

```diff
@@ -227,6 +227,10 @@ handleAction EvaluateActions =
             when (isSuccess result) do
               assign _lastEvaluatedSimulation simulation
               assign _blockchainVisualisationState Chain.initialState
+              -- preselect the first transaction (if any)
+              mAnnotatedBlockchain <- peruse (_evaluationResult <<< _Success <<< _Right <<< _resultRollup <<< to AnnotatedBlockchain)
```

Comment:
I think maybe it's time to pull out `_evaluationResult <<< _Success <<< _Right <<< _resultRollup`. At least to a local definition, if not to something in `MainFrame/Lenses.purs`.

I think the type signature will be something like `Prism' State (Array (Array AnnotatedTx))`.

Reply:
Good thinking. But is `_successfulEvaluationResult = _evaluationResult <<< _Success <<< _Right` maybe more natural? Even though we always happen to want the `resultRollup` at the moment. There's already a lens for `_successfulCompilationResult`, so that would match.